6/5 (#208)
dolthub/weekly-updates: emails/2026-06-05.md

This is the weekly CEO update from [DoltHub](https://www.dolthub.com/). I'm Tim, the CEO of DoltHub. 

We entertained newly formed [Gas City Inc](https://gascity.com/) in Santa Monica this week. [Beads](https://github.com/gastownhall/beads) is built on Dolt to expose version control features to agents. They made us honorary Polecats and we couldn't be happier. Expect more collaboration from our two teams moving forward.

### BranchBench

A team at Columbia University thinks [agents need branches](https://arxiv.org/abs/2604.17180). So much so that they proposed a new benchmark called BranchBench and measure Dolt and Neon on it. Obviously, we agree and were excited to [review their findings](https://www.dolthub.com/blog/2026-06-03-branch-bench-database-benchmarking-for-agentic-workflows/)! 

> Dolt did great on the version control part of the tasks but struggled on the query path for a couple queries. We identified two optimizations in our SQL engine and will have those patched ASAP. If you are trying Dolt and find a slow query, we will fix it! Dolt is built from the ground up so we have a ways to go before our SQL engine is optimization free. 

Reviewer: ```suggestion
Dolt did great on the version control part of the tasks but struggled on the query path for a couple queries. We identified two optimizations in our SQL engine and will have those patched ASAP. If you are trying Dolt and find a slow query, we will fix it! Dolt is built from the ground up so we have a ways to go before our SQL engine is fully optimized. 
```

[For simple queries we're faster than MySQL](https://www.dolthub.com/docs/sql-reference/benchmarks/latency/) so we know we can get there.

### Dumbo Garbage Collection

> We again implore you to let us know if your interested in version-controlled MongoDB. We're going to keep talking about it regardless. This week [Dumbo launched garbage collection](https://www.dolthub.com/blog/2026-06-02-dumbodb-gc/). As some of you old timers who've been using Dolt since before it had automated garbage collection know, Dolt's storage engine makes a lot of disk garbage. Garbage collection is essential for any moderate scale use case. Dumbo now has it. Bring us your use case.

Reviewer 2: ```suggestion
We again implore you to let us know if you're interested in version-controlled MongoDB. We're going to keep talking about it regardless. This week [Dumbo launched garbage collection](https://www.dolthub.com/blog/2026-06-02-dumbodb-gc/). As some of you old timers who've been using Dolt since before it had automated garbage collection know, Dolt's storage engine makes a lot of disk garbage. Garbage collection is essential for any moderate scale use case. Dumbo now has it. Bring us your use case.
```

### Dolt is for Agents in 300 Words

> Agents need version control. Dolt is is the only version-controlled SQL database. Dolt is the database for agents. [I expound on this topic in as few words as possible](https://www.dolthub.com/blog/2026-06-03-branch-bench-database-benchmarking-for-agentic-workflows/) trying to convince my fellow engineers.

Reviewer 2: ```suggestion
Agents need version control. Dolt is the only version-controlled SQL database. Dolt is the database for agents. [I expound on this topic in as few words as possible](https://www.dolthub.com/blog/2026-06-03-branch-bench-database-benchmarking-for-agentic-workflows/) trying to convince my fellow engineers.
```
